app/services/PrincipalConverter.scala

```diff
@@ -17,10 +17,11 @@
 package services
 
 import models.reference.Country
+import utils.StringTransformer._
 
 object PrincipalConverter extends Converter {
 
   def toViewModel(p: models.Principal, path: String, countries: Seq[Country]): ValidationResult[viewmodels.Principal] =
     findReferenceData(p.countryCode, countries, s"$path.countryCode")
-      .map(viewmodels.Principal(p.name, p.streetAndNumber, p.postCode, p.city, _, p.eori))
+      .map(viewmodels.Principal(p.name, p.streetAndNumber, p.streetAndNumber.shorten(32)("***"), p.postCode, p.city, _, p.eori, p.tir))
```

Comment:
Whats the purpose of the '***' bit, is it part of the formatting?

Reply:
The function strips it down to the length specified but masks the last 3 characters with ***. Included it as an argument in case we needed to use different characters elsewhere in the document